pages/profile/index.tsx

```diff
@@ -0,0 +1,283 @@
+import GitHubIcon from '@mui/icons-material/GitHub';
+import LinkedInImage from '@/public/icons/linkedin.png';
+import ChickenImage from '@/public/assets/profile-chicken-egg.png';
+import { TextField, TextFieldProps } from '@mui/material';
+import Link from 'next/link';
+import { RequestHelper } from '@/lib/request-helper';
+
+/**
+ * A page that allows a user to modify app or profile settings and see their data.
+ *
+ * Route: /profile
+ */
+export default function ProfilePage() {
+  const router = useRouter();
+  const { isSignedIn, hasProfile, user, profile, updateProfile } = useAuthContext();
+  const [uploading, setUploading] = useState<boolean>(false);
+  const resumeRef = useRef(null);
+
+  useEffect(() => {
+    if (hasProfile && !profile) window.location.reload();
+  }, [profile]);
+
+  const formatStudyLevel = (s: string) => {
+    if (s === 'grad') return 'Graduate Student';
+    return s.substring(0, 1).toUpperCase() + s.substring(1);
+  };
+
+  const formatRole = (s: string) => {
+    return s
+      .split('_')
+      .map((w) => w.substring(0, 1).toUpperCase() + w.substring(1))
+      .join(' ');
+  };
+
+  const textFieldOverrides: TextFieldProps = {
+    InputLabelProps: {
+      classes: {
+        root: '!text-black',
+      },
+    },
+    InputProps: {
+      classes: {
+        input: '!text-black [-webkit-text-fill-color:unset!important]',
+        notchedOutline: '!border-[#79747E]',
+      },
+    },
+  };
+
+  const handleResumeUpload = async (profile) => {
+    if (resumeRef.current.files.length !== 1) return alert('Must submit one file');
+
+    const fileExtension = getFileExtension(resumeRef.current.files[0].name);
+    const acceptedFileExtensions = [
+      '.pdf',
+      '.doc',
+      '.docx',
+      '.png',
+      '.jpg',
+      '.jpeg',
+      '.txt',
+      '.tex',
+      '.rtf',
+    ];
+
+    if (!acceptedFileExtensions.includes(fileExtension))
+      return alert(`Accepted file types: ${acceptedFileExtensions.join(' ')}`);
+
+    const resumeFile = resumeRef.current.files[0];
+
+    setUploading(true);
+
+    const formData = new FormData();
+    formData.append('resume', resumeFile);
+    formData.append('fileName', `${user.id}${fileExtension}`);
+    formData.append('studyLevel', profile.studyLevel);
+    formData.append('major', profile.major);
+
+    try {
+      const res = await fetch('/api/resume/upload', {
+        method: 'post',
+        body: formData,
+      });
+      // upload resume
+      const resumeUrl = (await res.json()).url;
+
+      // upload profile with new resume data
+      const { data } = await RequestHelper.put<
+        Registration,
+        { msg: string; registrationData: Registration }
+      >(
+        '/api/applications',
+        {},
+        {
+          ...profile,
+          resume: resumeUrl,
+        },
+      );
+      alert('Resume updated successfully');
+      updateProfile(data.registrationData);
+      window.location.reload();
```

Comment:
Any reason why we do this?

Reply:
after we update the profile, if we just do the router.push('/profile') it returns me undefined for profile, i think profile is not being fetched for some reason but the reload works.

Reply:
this is just to update the profile so that the resume changes is reflected